wagtail/admin/mail.py

```diff
@@ -328,11 +328,14 @@ def get_recipient_users(self, workflow_state, **kwargs):
 class BaseGroupApprovalTaskStateEmailNotifier(EmailNotificationMixin, Notifier):
     """A base notifier to send email updates for GroupApprovalTask events"""
 
+    task_classes = {GroupApprovalTask, }
+
     def __init__(self):
         super().__init__((TaskState,))
 
     def can_handle(self, instance, **kwargs):
-        if super().can_handle(instance, **kwargs) and isinstance(instance.task.specific, GroupApprovalTask):
+        if super().can_handle(instance, **kwargs) and instance.task.specific.__class__ in self.task_classes:
+            # do not use isinstance so subclasses of GroupApprovalTask can define their own Notifiers if wanted
```

Comment:
I'm in two minds about this - this means that if you subclass GroupApprovalTask for something really minor or unrelated to notifications, then you lose the notification functionality until you connect up a new signal handler for that subclass. This seems to go against the spirit of what object inheritance is for, and could be as much of a 'gotcha' as the double emails.

I'm not really sure how to fix that, though - it probably calls for a mechanism where each task subclass is only associated with a single notifier. Given that this change seems to be a bit tangential to the rest of this PR, would it be OK to leave this out for now and work out a better solution later?

Reply:
I agree, I couldn't think of a neat way to do this either. I've reverted for now - it may not be necessary to have this.